code/game/machinery/recharger.dm

```diff
@@ -34,142 +34,153 @@
+/obj/machinery/recharger/wrench_act(mob/living/user, obj/item/tool)
+	if(charging)
+		to_chat(user, span_notice("Remove the charging item first!"))
+		return TOOL_ACT_SIGNAL_BLOCKING
+	set_anchored(!anchored)
+	power_change()
+	to_chat(user, span_notice("You [anchored ? "attached" : "detached"] [src]."))
+	tool.play_tool_sound(src)
+	return TOOL_ACT_TOOLTYPE_SUCCESS
+
+/obj/machinery/recharger/screwdriver_act(mob/living/user, obj/item/tool)
+	if(!anchored || charging)
+		return TOOL_ACT_SIGNAL_BLOCKING
+	. = default_deconstruction_screwdriver(user, base_icon_state, base_icon_state, tool)
+	if(.)
+		update_appearance()
+
+/obj/machinery/recharger/crowbar_act(mob/living/user, obj/item/tool)
+	return (!anchored || charging) ? TOOL_ACT_SIGNAL_BLOCKING : default_deconstruction_crowbar(tool)
 
 /obj/machinery/recharger/attack_hand(mob/user, list/modifiers)
 	. = ..()
 	if(.)
 		return
 
 	add_fingerprint(user)
-	if(charging)
-		charging.update_appearance()
-		charging.forceMove(drop_location())
-		user.put_in_hands(charging)
-		setCharging(null)
-
+	if(!charging)
+		return
+	if(user.put_in_hands(charging))
+		return
+	charging.forceMove(drop_location())
 
 /obj/machinery/recharger/attack_tk(mob/user)
 	if(!charging)
 		return
-	charging.update_appearance()
 	charging.forceMove(drop_location())
-	setCharging(null)
 	return COMPONENT_CANCEL_ATTACK_CHAIN
 
-
 /obj/machinery/recharger/process(seconds_per_tick)
 	if(machine_stat & (NOPOWER|BROKEN) || !anchored)
 		return PROCESS_KILL
 
 	using_power = FALSE
-	if(charging)
-		var/obj/item/stock_parts/cell/C = charging.get_cell()
-		if(C)
-			if(C.charge < C.maxcharge)
-				C.give(C.chargerate * recharge_coeff * seconds_per_tick / 2)
-				use_power(active_power_usage * recharge_coeff * seconds_per_tick)
-				using_power = TRUE
-			update_appearance()
-
-		if(istype(charging, /obj/item/ammo_box/magazine/recharge))
-			var/obj/item/ammo_box/magazine/recharge/R = charging
-			if(R.stored_ammo.len < R.max_ammo)
-				R.stored_ammo += new R.ammo_type(R)
-				use_power(active_power_usage * recharge_coeff * seconds_per_tick)
-				using_power = TRUE
-			update_appearance()
-			return
-		if(!using_power && !finished_recharging) //Inserted thing is at max charge/ammo, notify those around us
-			finished_recharging = TRUE
-			playsound(src, 'sound/machines/ping.ogg', 30, TRUE)
-			say("[charging] has finished recharging!")
-
-	else
+	if(!charging)
 		return PROCESS_KILL
+	var/obj/item/stock_parts/cell/charging_cell = charging.get_cell()
+	if(charging_cell)
+		if(charging_cell.charge < charging_cell.maxcharge)
+			charging_cell.give(charging_cell.chargerate * recharge_coeff * seconds_per_tick / 2)
+			use_power(active_power_usage * recharge_coeff * seconds_per_tick)
+			using_power = TRUE
+		update_appearance()
+
+	if(istype(charging, /obj/item/ammo_box/magazine/recharge)) //if you add any more snowflake ones, make sure to update the examine messages too.
+		var/obj/item/ammo_box/magazine/recharge/R = charging
```

Comment:
if you're gonna be culling single-letter variables in this refactor you may as well get them all.